src/app/modules/main-module/components/main/main.component.ts

```diff
@@ -76,12 +89,21 @@ export class MainComponent implements OnInit {
   }
 
   writeLetter(tecla: string) {
-    this.word[this.positionInput] = tecla;
-    this.positionInput = this.findCorrectIndex();
+    this.word[this.positionRoundLetter] = tecla;
+    this.positionRoundLetter = this.findCorrectIndex();
+  }
+
+  getPosition(idCasilla: number, idRound: number) {
+    if(idRound === this.contRound){
```

Comment:
los if es mejor poner la condición que no quieres que se de para tu asignación y ponerle un return, de esa manera la función no recorre código innecesariamente  if(idRound !== this.contRound) return